ouroboros-consensus/src/Ouroboros/Consensus/Util/STM.hs

```diff
@@ -68,6 +84,11 @@ blockUntilJust getMaybeA = do
 blockUntilAllJust :: MonadSTM m => [STM m (Maybe a)] -> STM m [a]
 blockUntilAllJust = mapM blockUntilJust
 
+-- | Simple type that can be used to indicate something in a 'TVar' is
+-- changed.
+newtype Fingerprint = Fingerprint Word64
```

Comment:
Will this be confusing with fingerprints used in the network layer? (Don't know.)

Reply:
In the network layer, there's only `FetchStateFingerprint`, so i don't think there's a risk of confusion.